src/frontend/components/Felleskomponenter/JaNeiSpm/JaNeiSpmForSanity.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { ReactNode, useEffect, useState } from 'react';
+
+import { v4 as uuidv4 } from 'uuid';
+
+import { Box } from '@navikt/ds-react';
+import { ESvar, JaNeiSpørsmål } from '@navikt/familie-form-elements';
+import { Felt, ISkjema } from '@navikt/familie-skjema';
+
+import { useApp } from '../../../context/AppContext';
+import { AlternativtSvarForInput } from '../../../typer/common';
+import { FlettefeltVerdier, ISanitySpørsmålDokument } from '../../../typer/sanity/sanity';
+import { SkjemaFeltTyper } from '../../../typer/skjema';
+import { logSpørsmålBesvart } from '../../../utils/amplitude';
+import TekstBlock from '../Sanity/TekstBlock';
+
+interface IJaNeiSpmProps {
+    skjema: ISkjema<SkjemaFeltTyper, string>;
+    felt: Felt<ESvar | null>;
+    tilleggsinfo?: ReactNode;
+    inkluderVetIkke?: boolean;
+    spørsmålDokument: ISanitySpørsmålDokument;
+    flettefelter?: FlettefeltVerdier;
+}
+
+const JaNeiSpm: React.FC<IJaNeiSpmProps> = ({
+    skjema,
+    felt,
+    tilleggsinfo,
+    inkluderVetIkke = false,
+    spørsmålDokument,
+    flettefelter,
+}) => {
+    const [mounted, settMounted] = useState(false);
+    const { søknad, tekster, plainTekst } = useApp();
+    const { ja, nei, jegVetIkke } = tekster().FELLES.frittståendeOrd;
+
+    useEffect(() => {
+        if (mounted) {
+            spørsmålDokument &&
+                logSpørsmålBesvart(
+                    spørsmålDokument.api_navn,
+                    felt.verdi ?? AlternativtSvarForInput.UKJENT,
+                    søknad.søknadstype
+                );
+        }
+        settMounted(true);
+    }, [felt.verdi]);
```

Comment:
Gjennom tidene har `useEffect` blitt misbrukt og slik som det er nå blir det som regel ansatt som et anti-pattern i de fleste tilfeller. https://react.dev/learn/you-might-not-need-an-effect denne er verdt å ta en titt på. Jeg ser ikke umiddelbart at dette ikke er en plass hvor `useEffect` burde brukes, men det kan fort hende at denne logikken egentlig burde leve en annen plass

Reply:
Jeg bytter fra bruk av useEffect til onChange 👍

Reply:
Oppretter et eget favrokort på dette https://favro.com/organization/98c34fb974ce445eac854de0/1844bbac3b6605eacc8f5543?card=NAV-22845